cli/testcomposer/testcomposer.go

```diff
@@ -0,0 +1,74 @@
+package testcomposer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"io/ioutil"
+	"net/http"
+	"os"
+)
+
+// Client service
+type Client struct {
+	HTTPClient http.Client
+	URL        string // https://api.staging.saucelabs.net
+}
+
+// Job represents the sauce labs test job.
+
+type Job struct {
+	ID    string `json:"id"`
+	Owner string `json:"owner"`
+}
+
+// JobStarterPayload is a JSON object of parameters used to start a session
+// from saucectl
+type JobStarterPayload struct {
+	User        string   `json:"username"`
+	AccessKey   string   `json:"accessKey"`
+	BrowserName string   `json:"browserName,omitempty"`
+	TestName    string   `json:"testName,omitempty"`
+	Framework   string   `json:"framework,omitempty"`
+	BuildName   string   `json:"buildName,omitempty"`
+	Tags        []string `json:"tags,omitempty"`
+}
+
+func (c *Client) StartJob(ctx context.Context, jobStarterPayload JobStarterPayload) (jobID string, err error) {
+	url := fmt.Sprintf("%s/v1/testcomposer/jobs/", c.URL)
+	b := new(bytes.Buffer)
+	err = json.NewEncoder(b).Encode(jobStarterPayload)
+	if err != nil {
+		return
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, b)
+	if err != nil {
+		return
+	}
+	req.SetBasicAuth(os.Getenv("SAUCE_USERNAME"), os.Getenv("SAUCE_ACCESS_KEY"))
```

Comment:
@alexplischke the credentials won't be coming from a payload, this will be running on the user's machine and the creds need to be sent to test-composer.

Reply:
`jobStarterPayload` has `User` and `AccessKey` right? And those will have to be populated.
Reading env variables inside the client isn't obvious to the caller of method `StartJob()`.